### DISCUSSION TITLE
Issues playing cached video files with and without rangeRequests

### DISCUSSION THREAD

**Discussion by scottfuoco** [serwist/serwist#85] [Q&A]
Hi there, I'm trying to use a CacheFirst strategy for serving video assets.  The videos play as expected when there is a cache miss and the video gets requested from the network.  However, once the videos are in the cache the video doesn't play, instead it hangs on a blank video player with a loading spinner.  I'm using AWS S3 and Cloudfront as my CDN, and I've been jumping around posts trying to see if there are any security / permission restrictions that would cause issues.

The runtime caching strategy I'm using looks like this

```
    {
      urlPattern: /.*\.(mp4|mov|webm)$/i,
      handler: "CacheFirst",
      options: {
        cacheName: "video-cache",
        expiration: {
          maxAgeSeconds: ONE_DAY_IN_SECONDS,
        },
        plugins: [
          new CacheableResponsePlugin({
            statuses: [0, 200],
          }),
        ],
        rangeRequests: true,
      },
    }
```

I need CacheableResponsePlugin to catch statuses of 0 because of "opaque" video responses coming back from my Cloudfront CDN.  I don't know for sure if this could be the cause of my issues, but from what I have found online this shouldn't be a problem as long as the video ends up in the cache.

I have created simple demo repo with my Serwist setup/config and one of the videos that are having the an issue once it has been cached here

[https://github.com/scottfuoco/serwist-video-cache](https://github.com/scottfuoco/serwist-video-cache)

Thanks for any help/guidance!

**Comment by DuCanhGH** [comment]
@scottfuoco  the demo seems to work just fine to me though. It returned the 206 status code (a partial response), so it wasn't cached. Probably not what you were looking for.

Oh and here's a bit of a guide from Workbox: https://developer.chrome.com/docs/workbox/serving-cached-audio-and-video. Hope this helps!

~~I think that your problem as described in the details is not related to the guide and its attached issue though. But I wouldn't know what that is about until I can see it...~~ (edit: I'd like to take back what I said. Seems like it is the same issue, even)
